modules/packages/Yaml.chpl

```diff
@@ -0,0 +1,1935 @@
+/*
+ * Copyright 2020-2023 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+
+Support for parsing and writing YAML files.
+
+Includes a Yaml Serializer/Deserializer as well as an abstract class
+representation of a YAML file.
+
+The Yaml Serializer/Deserializer is intended to be used with the standard
+IO module's serialization/deserialization API. For example:
+
+.. code-block:: chpl
+
+  use Yaml;
+
+  record R {
+    var a: int;
+    var b: string;
+  }
+
+  var myFile = open("r.yaml", iomode.cwr),
+      r1 = new R(1, "hello");
+
+  myFile.writer().withSerializer(new YamlSerializer()).write(r1);
```

Comment:
FWIW you should be able to write ``myFile.writer(serializer=new YamlSerializer()).write(r1);`` instead.